react-app/src/components/ProposalScreen/ProposalScreen.tsx

```diff
@@ -54,10 +54,7 @@ const ProposalScreen: React.FC = () => {
   );
   const [after, setAfter] = useState(0);
 
-  const [searchParams, setSearchParams] = useSearchParams({
-    after: after.toString(),
-    tab: defaultTabItem.value,
-  });
+  const [searchParams, setSearchParams] = useSearchParams();
```

Comment:
please treat `useSearchParams` as an aggregated `useState`, the params should reflect the actual state instead of a copy of the state or else you will get a desync between the states at some point

```
const [searchParams, setSearchParams] = useSearchParams({
    page: 1,
    tab: defaultTabItem.value,
});

const [after, tab] = useMemo(() => {
    const after = (parseInt(page, 10) - 1) * PROPOSAL_LIST_PAGE_SIZE
    const tab = tabItems.find((item) => item.value === searchParams.get("tab"))
    return [after, tab];
},[searchParams])

const setPage = useCallback((after: number) => {
    setSearchParams({
        ...searchParams,
        page: (after / PROPOSAL_LIST_PAGE_SIZE + 1).toString()
    })
},[searchParams]);

const handleSelectTab = useCallback((tab: FilterKey) => {
    setSearchParams({
        ...searchParams,
        tab
    })
},[searchParams])
```